exploration/selection-declaration.md

```diff
@@ -0,0 +1,296 @@
+declaration       = s variable [s] "=" [s] expression
+```
+
+**Pros**
+- Shorthand is consistent with the rest of the syntax
+- Shorthand version works intuitively with minimal input
+- Preserves immutability
+- Produces an error when users inappropriately annotate some items
+
+**Cons**
+- Selectors can't provide additional selection-specific options
+  if the value has already been annotated
+- Doesn't allow multiple selection on the same operand, e.g.
+  ```
+  .input {$date :datetime skeleton=yMdjm}
+  .match {$date :datetime field=month} {$date :datetime field=dayOfWeek}
+  * * {{This message produces a Duplicate Declaration error
+        even though selection is separate from formatting.}}
+  ```
+  However, this design does allow for a local variable to be easily created
+  for the purpose of selection.
+
+#### Allow "Declarative" Selectors with _Mutability_
+
+In this alternative, selectors are treated as declaration-selectors.
+That is, an annotation in a selector works like a `.input`.
+This permits `.match` selectors to be a shorthand when no declarations exist.
+
+It is not an error to overwrite a previous annotation.
+Instead the selector's annotation replaces what came before.
+
+```
+.input {$num :integer}
+.match {$num :number minimumFractionDigits=1}
+* {{Formats {$num} like 1.0}}
+```
+
+**Pros**
+- Shorthand version works intuitively with minimal typing.
+
+**Cons**
+- Violates immutability that we've established everywhere else
+
+#### Allow "Declarative" Selectors with _Immutability_
+
+In this alternative, selectors are treated as declaration-selectors.
+That is, an annotation in a selector works like a `.input`.
+However, it is an error for the selector to try to modify a previous declaration
+(just as it is an error for a declaration to try to modify a previous declaration).
+This permits `.match` selectors to be a shorthand when no declarations exist.
+
+It is also an error for a selector to modify a previous selector.
+This implies that multiple selecton on the same operand is pointless.
+
+```
+.match {$num :integer}
+* {{Formats {$num} as integer}}
+
+.input {$num :integer}
+.match {$num :number maximumFractionDigits=0}
+* {{This message produces a Duplicate Declaration error}}
+
+.input {$num :integer} {$num :number}
+* * {{This message produces a Duplicate Declaration error}}
+```
+
+**Pros**
+- Shorthand version works intuitively with minimal typing
+- Preserves immutability
+- Produces an error when users inappropriately annotate some items
+
+**Cons**
+- Selectors can't provide additional selection-specific options
+  if the value has already been annotated
+- Doesn't allow multiple selection on the same operand, e.g.
+  ```
+  .input {$date :datetime skeleton=yMdjm}
+  .match {$date :datetime field=month} {$date :datetime field=dayOfWeek}
+  * * {{This message produces a Duplicate Declaration error
+        even though selection is separate from formatting.}}
+  ```
+
```

Comment:
Let's add my proposal from https://github.com/unicode-org/message-format-wg/issues/736#issuecomment-2016408105:

````suggestion

#### Match on variables instead of expressions

In this alternative, the `.match` syntax is simplified
to work on variable references rather than expressions:

```
.input {$count :number}
.match $count
one {{You have {$count} apple}}
* {{You have {$count} apples}}

.local $empty = {$theList :isEmpty}
.match $empty
true {{You bought nothing}}
* {{You bought {$theList}!}}
```

The ABNF change would look like:
```diff
 match-statement   = match 1*([s] selector)
-selector          = expression
+selector          = variable
```


**Pros**
- Overall the syntax is simplified.
- Preserves immutability.

**Cons**
- A separate declaration is required for each selector.

````